code/game/objects/structures/aliens.dm

```diff
@@ -239,44 +243,46 @@
 				return
 			if(GROWN)
 				to_chat(user, "<span class='notice'>You retrieve the child.</span>")
-				Burst(0)
+				burst(FALSE)
 				return
 	else
 		to_chat(user, "<span class='notice'>It feels slimy.</span>")
 		user.changeNext_move(CLICK_CD_MELEE)
 
 
-/obj/structure/alien/egg/proc/GetFacehugger()
+/obj/structure/alien/egg/proc/getFacehugger()
 	return locate(/obj/item/clothing/mask/facehugger) in contents
 
-/obj/structure/alien/egg/proc/Grow()
+/obj/structure/alien/egg/proc/grow()
 	icon_state = "egg"
 	status = GROWN
 	AddComponent(/datum/component/proximity_monitor)
 
-/obj/structure/alien/egg/proc/Burst(kill = TRUE)	//drops and kills the hugger if any is remaining
+///Need to carry the kill from Burst() to Hatch(), this section handles the alien opening the egg
+/obj/structure/alien/egg/proc/burst(kill)
 	if(status == GROWN || status == GROWING)
 		icon_state = "egg_hatched"
 		flick("egg_opening", src)
 		status = BURSTING
 		qdel(GetComponent(/datum/component/proximity_monitor))
-		spawn(15)
-			status = BURST
-			var/obj/item/clothing/mask/facehugger/child = GetFacehugger()
-			if(child)
-				child.loc = get_turf(src)
-				if(kill && istype(child))
-					child.Die()
-				else
-					for(var/mob/M in range(1,src))
-						if(CanHug(M))
-							child.Attach(M)
-							break
+		addtimer(CALLBACK(src, /obj/structure/alien/egg/.proc/hatch), 1.5 SECONDS)
+
+///We now check HOW the hugger is hatching, kill carried from Burst() and obj_break()
+/obj/structure/alien/egg/proc/hatch(kill)
+	status = BURST
+	var/obj/item/clothing/mask/facehugger/child = getFacehugger()
+	child.forceMove(get_turf(src))
+	if(kill)
+		child.Die()
+	for(var/mob/M in range(1, src))
+		if(CanHug(M))
+			child.Attach(M)
+			break
```

Comment:
Why remove the else for the for loop? If the hugger is dying, shouldn't we not want it hugging people?

Reply:
> Why remove the else for the for loop? If the hugger is dying, shouldn't we not want it hugging people?

Eggs delete huggers on death, this would never be able to occur in game
I assume it's trying to do "if the huggers is not dead, then if a target is in range let them hatch"